cmd/medsky/bgs/bgs.go

```diff
@@ -0,0 +1,1440 @@
+package bgs
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/bluesky-social/indigo/atproto/identity"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+	"github.com/ipfs/go-cid"
+	"log/slog"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	comatproto "github.com/bluesky-social/indigo/api/atproto"
+	"github.com/bluesky-social/indigo/cmd/medsky/events"
+	"github.com/bluesky-social/indigo/cmd/medsky/repomgr"
+	lexutil "github.com/bluesky-social/indigo/lex/util"
+	"github.com/bluesky-social/indigo/models"
+	"github.com/bluesky-social/indigo/xrpc"
+
+	"github.com/gorilla/websocket"
+	lru "github.com/hashicorp/golang-lru/v2"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	promclient "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"gorm.io/gorm"
+)
+
+var tracer = otel.Tracer("bgs")
+
+// serverListenerBootTimeout is how long to wait for the requested server socket
+// to become available for use. This is an arbitrary timeout that should be safe
+// on any platform, but there's no great way to weave this timeout without
+// adding another parameter to the (at time of writing) long signature of
+// NewServer.
+const serverListenerBootTimeout = 5 * time.Second
+
+type BGS struct {
+	db      *gorm.DB
+	slurper *Slurper
+	events  *events.EventManager
+	didd    identity.Directory
```

Comment:
I usually use `dir` as the short variable name for directories; could reduce confusion with the old did directory code to update it?